File: Python/27.Remove Element.py
#从数组中删掉某元素,返回删掉后的数组长度
class Solution(object):
    def removeElement(self, nums, val):
        size = 0
        for i in range(len(nums)):
            if nums[i]!=val:
                nums[size] = nums[i]
                size += 1
        return size

s = Solution()
print(s. removeElement([1,1,2,2,3],1))

File: C++/Two Pointers/27.Remove Element.cpp
/*
将数组中的给定元素删除，返回删除后的长度。
*/

#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
     int removeElement(vector<int>& nums, int val) {
        int length = nums.size(),pos = 0;
        for(int n:nums)
        {
        	if(n!=val)	nums[pos++] = n;
        }
        return pos;
    }
};

int main()
{
	Solution s;
	vector<int> nums{3,2,2,3};
	cout<<s.removeElement(nums, 3)<<endl;
}

